src/core/apiLayer/productService.ts

```diff
@@ -0,0 +1,125 @@
+import { stringify } from "query-string";
+import { useEffect, useReducer } from "react";
+
+import { apiUrl } from "@temp/constants";
+
+const fetchProducts = (criteria: any, limit: number, offset: number) => {
+  const query = stringify({
+    ...criteria,
+    currency: "EUR",
+    limit,
+    offset,
+  });
+
+  return fetch(`${apiUrl}../pkb/products/?${query}`, {
+    headers: {
+      Authorization: "Basic YWRtaW5AYWRtaW4uY29tOm9NWFVvR1p0UDV6Mw==",
+    },
+  }).then(response => response.json());
+};
+
+const initialState = {
+  productList: [],
+  offset: 0,
+  todoOffsets: [],
+  criteria: {},
+  loading: true,
+  hasMore: false,
+  pageSize: 24,
+};
+
+function reducer(state, action) {
+  switch (action.type) {
+    case "changeCriteria":
+      if (Object.is(state.criteria, action.criteria)) {
+        return state;
+      }
+
+      return {
+        ...state,
+        criteria: action.criteria,
+        offset: 0,
+        todoOffsets: [],
```

Comment:
What does this array do? Is it the set of "load more" requests piled up?

Reply:
yes, I am open for a better name if it is not clear.

Reply:
What about `offsetQueue`? Not quite sure.

Reply:
The first name I choose was exactly that :D